sormas-ui/src/main/java/de/symeda/sormas/ui/utils/FilteredGrid.java

```diff
@@ -256,10 +256,10 @@ public void bulkActionHandler(Consumer<Set> callback, boolean allowAdminOverride
 	/**
 	 * @deprecated Use {@link #getDataSize()}, because this call triggers another count query for lazy {@link DataProvider}s.
 	 */
-	@Deprecated
-	public int getItemCount() {
-		return getDataProvider().size(new Query<>());
-	}
+//	@Deprecated
```

Comment:
The changes you mentioned to points 1 and 2 are meant to keep the existing behavior in the Infrastructure menu and fix some bugs also. There was an issue regarding bulkEdit mode which has not been noticed so far. If you look at how the bulkEdit is implemented you can see that in bulkEdit mode the search field should be disabled. Well, this works only exactly when you press the "EnterBulkEditMode" button. The field does not stay disabled once another dropdown is changed or you go in and out of the form (cache-related issue). Since I was working here and while during tests I felt they are annoying I also looked to fix them. The part that the searchField is cleared once you enter the bulkEdit mode comes from me. I felt it is the right behavior. If you don't do this the other dropdowns which remain available in the bulkEdit mode are almost useless. This can be removed but I would not recommend it. 
The method getItemCount() has been removed.